models/models.js

```diff
@@ -21,10 +21,19 @@ exports.fetchArticleById = (article_id) => {
     });
 };
 
-exports.fetchArticles = () => {
+exports.fetchArticles = (sort_by = "created_at", order = "desc") => {
+  const validSortBys = ["title", "topic", "author", "body", "created_at"];
+  const validOrders = ["asc", "desc"];
+
+  if (!validSortBys.includes(sort_by)) {
+    return Promise.reject({ status: 400, message: "bad request" });
+  }
+  if (!validOrders.includes(order)) {
+    return Promise.reject({ status: 400, message: "bad request" });
+  }
   return db
     .query(
-      "SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT (comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY created_at DESC"
+      `SELECT articles.article_id, articles.title, articles.author, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT (comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.article_id ORDER BY ${sort_by} ${order}`
```

Comment:
To improve readability, you can select all columns like this: `SELECT articles.*,` 💡